### ISSUE TITLE
User module fails when user already present

### ISSUE THREAD

**Issue by codingonHP** [ansible/ansible#63563]
##### SUMMARY
When executing user creation ansible module, below error is seen consistently.
Sample playbook and its corresponding output can be seen below

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
system/user module

##### ANSIBLE VERSION

```
ansible 2.8.5
  config file = /home/ansible_user/ansible-repo/ansible.cfg
  configured module search path = [u'/home/ansible_user/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.5 (default, May 31 2018, 09:41:32) [GCC 4.8.5 20150623 (Red Hat 4.8.5-28)]
```

##### CONFIGURATION

```DEFAULT_HOST_LIST(/home/ansible_user/ansible-repo/ansible.cfg) = [u'/home/ansible_user/.ansible_protected/inventory', u'/home/ansible_user/ansible-repo/environments/common',
DEFAULT_TIMEOUT(/home/ansible_user/ansible-repo/ansible.cfg) = 30
DEFAULT_VAULT_PASSWORD_FILE(/home/ansible_user/ansible-repo/ansible.cfg) = /home/ansible_user/.ansible_protected/ansible_vault_pass
PERSISTENT_COMMAND_TIMEOUT(/home/ansible_user/ansible-repo/ansible.cfg) = 30
PERSISTENT_CONNECT_TIMEOUT(/home/ansible_user/ansible-repo/ansible.cfg) = 30
```

##### OS / ENVIRONMENT
RHEL 7.4

##### STEPS TO REPRODUCE

```yaml
### Users / Groups
---
  - name: Testing user module
    hosts: all
    tasks:
    - name: Add primary group
      group:
          name: "some_user"
          state: present

    - name: Add user
      user:
          name: "some_user"
          comment: "service user"
          shell: "/bin/bash"
          home: "/opt/some_user"
          state: present

```

##### EXPECTED RESULTS
The user should not fail when user is state=present and user already present with expected properties

##### ACTUAL RESULTS
```
[ansible_user@ansible_host ansible-repo]$ ansible-playbook test.yml --limit host-3
 [WARNING]: Unable to parse /home/ansible_user/.ansible_protected/inventory as an inventory source

 [WARNING]: Unable to parse /home/ansible_user/ansible-repo/environments/common as an inventory source


PLAY [Testing user module] ***********************************************************************************************************************************

TASK [Gathering Facts] ***************************************************************************************************************************************
ok: [host-3]

TASK [Add primary group] *************************************************************************************************************************************
ok: [host-3]

TASK [Add user] **********************************************************************************************************************************************
fatal: [host-3]: FAILED! => {"changed": false, "msg": "usermod: user some_user is currently used by process 5003\n", "name": "some_user", "rc": 8}

PLAY RECAP ***************************************************************************************************************************************************
host-3                     : ok=2    changed=0    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0

[ansible_user@ansible_host ansible-repo]$

```

**Comment by ansibot** [comment]
Files identified in the description:
* [`lib/ansible/module_utils/facts/system/user.py`](https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/facts/system/user.py)
* [`lib/ansible/modules/system/user.py`](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/system/user.py)

If these files are inaccurate, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by samdoran** [comment]
I am unable to duplicate this issue. Can you provide the output of the playbook run with `-vvvv`? This looks like a problem with the remote system. Can you investigate and see what process 5003 is doing?

needs_info

**Comment by codingonHP** [comment]
Hi @samdoran 
Here is the command output with -vvvvv 
The process 5003  is  `ssh-agent` that is running on the target server.
We had a question on why usermod command getting executed when user is already present.
Shouldn't the task be skipped ?

### Output
``` bash


TASK [pkg-shared : Add user some_user with group some_user  }}] *********************************************************************************************************************************
task path: /home/user/ansible/roles/pkg-shared/tasks/util/user.yml:8
<10.X.X.51> ESTABLISH PARAMIKO SSH CONNECTION FOR USER: user on PORT 22 TO 10.X.X.51
<10.X.X.53> ESTABLISH PARAMIKO SSH CONNECTION FOR USER: user on PORT 22 TO 10.X.X.53
<10.X.X.51> EXEC /bin/sh -c 'echo ~user && sleep 0'
<10.X.X.51> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /home/user/.ansible/tmp/ansible-tmp-1571285921.29-253635874554425 `" && echo ansible-tmp-1571285921.29-253635874554425="` echo /home/user/.ansible/tmp/ansible-tmp-1571285921.29-253635874554425 `" ) && sleep 0'
<10.X.X.53> EXEC /bin/sh -c 'echo ~user && sleep 0'
<10.X.X.53> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /home/user/.ansible/tmp/ansible-tmp-1571285921.6-220680369715941 `" && echo ansible-tmp-1571285921.6-220680369715941="` echo /home/user/.ansible/tmp/ansible-tmp-1571285921.6-220680369715941 `" ) && sleep 0'
Using module_utils file /usr/lib/python2.7/site-packages/ansible/module_utils/_text.py
Using module_utils file /usr/lib/python2.7/site-packages/ansible/module_utils/basic.py
Using module_utils file /usr/lib/python2.7/site-packages/ansible/module_utils/distro/__init__.py
Using module_utils file /usr/lib/python2.7/site-packages/ansible/module_utils/six/__init__.py
Using module_utils file /usr/lib/python2.7/site-packages/ansible/module_utils/parsing/convert_bool.py
Using module_utils file /usr/lib/python2.7/site-packages/ansible/module_utils/common/__init__.py
Using module_utils file /usr/lib/python2.7/site-packages/ansible/module_utils/common/_collections_compat.py
Using module_utils file /usr/lib/python2.7/site-packages/ansible/module_utils/common/validation.py
Using module_utils file /usr/lib/python2.7/site-packages/ansible/module_utils/common/sys_info.py
Using module_utils file /usr/lib/python2.7/site-packages/ansible/module_utils/parsing/__init__.py
Using module_utils file /usr/lib/python2.7/site-packages/ansible/module_utils/common/text/__init__.py
Using module_utils file /usr/lib/python2.7/site-packages/ansible/module_utils/pycompat24.py
Using module_utils file /usr/lib/python2.7/site-packages/ansible/module_utils/common/_utils.py
Using module_utils file /usr/lib/python2.7/site-packages/ansible/module_utils/common/process.py
Using module_utils file /usr/lib/python2.7/site-packages/ansible/module_utils/common/text/formatters.py
Using module_utils file /usr/lib/python2.7/site-packages/ansible/module_utils/common/file.py
Using module_utils file /usr/lib/python2.7/site-packages/ansible/module_utils/common/text/converters.py
Using module_utils file /usr/lib/python2.7/site-packages/ansible/module_utils/common/parameters.py
Using module_utils file /usr/lib/python2.7/site-packages/ansible/module_utils/common/_json_compat.py
Using module_utils file /usr/lib/python2.7/site-packages/ansible/module_utils/common/collections.py
Using module_utils file /usr/lib/python2.7/site-packages/ansible/module_utils/distro/_distro.py
Using module file /usr/lib/python2.7/site-packages/ansible/modules/system/user.py
<10.X.X.51> PUT /home/user/.ansible/tmp/ansible-local-10565UlAd4j/tmp0PeTd9 TO /home/user/.ansible/tmp/ansible-tmp-1571285921.29-253635874554425/AnsiballZ_user.py
Using module file /usr/lib/python2.7/site-packages/ansible/modules/system/user.py
<10.X.X.53> PUT /home/user/.ansible/tmp/ansible-local-10565UlAd4j/tmpNsxa35 TO /home/user/.ansible/tmp/ansible-tmp-1571285921.6-220680369715941/AnsiballZ_user.py
<10.X.X.51> EXEC /bin/sh -c 'chmod u+x /home/user/.ansible/tmp/ansible-tmp-1571285921.29-253635874554425/ /home/user/.ansible/tmp/ansible-tmp-1571285921.29-253635874554425/AnsiballZ_user.py && sleep 0'
<10.X.X.53> EXEC /bin/sh -c 'chmod u+x /home/user/.ansible/tmp/ansible-tmp-1571285921.6-220680369715941/ /home/user/.ansible/tmp/ansible-tmp-1571285921.6-220680369715941/AnsiballZ_user.py && sleep 0'
<10.X.X.53> EXEC /bin/sh -c 'sudo -H -S  -p "[sudo via ansible, key=kefbxwmfrylocrdfwupgbpbujsmmnfxu] password:" -u root /bin/sh -c '"'"'echo BECOME-SUCCESS-kefbxwmfrylocrdfwupgbpbujsmmnfxu ; /usr/bin/python /home/user/.ansible/tmp/ansible-tmp-1571285921.6-220680369715941/AnsiballZ_user.py'"'"' && sleep 0'
<10.X.X.51> EXEC /bin/sh -c 'sudo -H -S  -p "[sudo via ansible, key=pqmhyywonfkznzbmjuofwthgvbtklgqq] password:" -u root /bin/sh -c '"'"'echo BECOME-SUCCESS-pqmhyywonfkznzbmjuofwthgvbtklgqq ; /usr/bin/python /home/user/.ansible/tmp/ansible-tmp-1571285921.29-253635874554425/AnsiballZ_user.py'"'"' && sleep 0'
<10.X.X.53> EXEC /bin/sh -c 'rm -f -r /home/user/.ansible/tmp/ansible-tmp-1571285921.6-220680369715941/ > /dev/null 2>&1 && sleep 0'
<10.X.X.51> EXEC /bin/sh -c 'rm -f -r /home/user/.ansible/tmp/ansible-tmp-1571285921.29-253635874554425/ > /dev/null 2>&1 && sleep 0'
fatal: [host-3]: FAILED! => {
    "changed": false,
    "invocation": {
        "module_args": {
            "append": false,
            "authorization": null,
            "comment": "user for services",
            "create_home": true,
            "expires": null,
            "force": false,
            "generate_ssh_key": null,
            "group": null,
            "groups": null,
            "hidden": null,
            "home": "/opt/some_user",
            "local": null,
            "login_class": null,
            "move_home": false,
            "name": "some_user",
            "non_unique": false,
            "password": null,
            "password_lock": null,
            "profile": null,
            "remove": false,
            "role": null,
            "seuser": null,
            "shell": "/bin/bash",
            "skeleton": null,
            "ssh_key_bits": 0,
            "ssh_key_comment": "ansible-generated on server",
            "ssh_key_file": null,
            "ssh_key_passphrase": null,
            "ssh_key_type": "rsa",
            "state": "present",
            "system": false,
            "uid": null,
            "update_password": "always"
        }
    },
    "msg": "usermod: user some_user is currently used by process 5003\n",
    "name": "some_user",
    "rc": 8
}
fatal: [host-2]: FAILED! => {
    "changed": false,
    "invocation": {
        "module_args": {
            "append": false,
            "authorization": null,
            "comment": "user for  services",
            "create_home": true,
            "expires": null,
            "force": false,
            "generate_ssh_key": null,
            "group": null,
            "groups": null,
            "hidden": null,
            "home": "/opt/some_user",
            "local": null,
            "login_class": null,
            "move_home": false,
            "name": "some_user",
            "non_unique": false,
            "password": null,
            "password_lock": null,
            "profile": null,
            "remove": false,
            "role": null,
            "seuser": null,
            "shell": "/bin/bash",
            "skeleton": null,
            "ssh_key_bits": 0,
            "ssh_key_comment": "ansible-generated on server",
            "ssh_key_file": null,
            "ssh_key_passphrase": null,
            "ssh_key_type": "rsa",
            "state": "present",
            "system": false,
            "uid": null,
            "update_password": "always"
        }
    },
    "msg": "usermod: user some_user is currently used by process 13153\n",
    "name": "some_user",
    "rc": 8
}

PLAY RECAP ********************************************************************************************************************************************************************************
host-1                     : ok=2    changed=0    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0
host-2                     : ok=14   changed=0    unreachable=0    failed=1    skipped=7    rescued=0    ignored=1
host-3                     : ok=14   changed=0    unreachable=0    failed=1    skipped=7    rescued=0    ignored=1
localhost                  : ok=1    changed=0    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0


```

**Comment by samdoran** [comment]
The only reason `usermod` would be run when an account already exists is if the account on the system doesn't match what is in the task.

Completely remove the account from the system, then run the task only once and see if the user name, comment, home, and shell match the task. If they do not, then something is preventing the `useradd` command from properly creating the account.

Then, try removing `comment, `shell`, and `home` one at a time to see which option is triggering a modification.

needs_info

**Comment by codingonHP** [comment]
@samdoran only account name should match or all of the parameters like shell, home should match?

**Comment by samdoran** [comment]
All parameters should match. If anything is different, then `usermod` will be run in an attempt to bring to account state into alignment.

needs_info

**Comment by ansibot** [comment]
@codingonHP This issue is waiting for your response. Please respond or the issue will be closed.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@codingonHP You have not responded to information requests in this issue so we will assume it no longer affects you. If you are still interested in this, please create a new issue with the requested information.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)
